httpclient5/src/main/java/org/apache/hc/client5/http/psl/PublicSuffixMatcher.java

```diff
@@ -118,20 +118,47 @@ private static boolean match(final DomainType domainType, final DomainType expec
      * @return domain root
      */
     public String getDomainRoot(final String domain) {
-        return getDomainRoot(domain, null);
+        return getDomainRoot(domain, null, false);
     }
 
     /**
      * Returns registrable part of the domain for the given domain name or {@code null}
      * if given domain represents a public suffix.
      *
      * @param domain
-     * @param expectedType expected domain type or {@code null} if any.
+     * @param legacyMode set to {@code true} to preserve behaviour prior to version 5.4.
      * @return domain root
+     * @since 5.4
+     */
+    public String getDomainRoot(final String domain, final boolean legacyMode) {
+        return getDomainRoot(domain, null, legacyMode);
+    }
+
+    /**
+     * Returns registrable part of the domain for the given domain name or {@code null}
+     * if given domain represents a public suffix. For backwards-compatability reasons this uses
+     * "legacy mode" to retain previous behaviour.
```

Comment:
This constructor currently sets `legacyMode` to `false`

Reply:
Yes, so we actually default to the "correct" behaviour. I'll update the comment here to reflect that.